Packs/CrowdStrikeFalconX/Integrations/CrowdStrikeFalconX/CrowdStrikeFalconX.py

```diff
@@ -430,95 +439,174 @@ def find_submission_id(
     :param sandbox_fields: the wanted params that appear in the sandbox section
     :param extra_sandbox_fields: the wanted params that appear in the extra sandbox section
-    :return: a dict based on api_res with the wanted params only
     """
-    api_res_quota, api_res_resources, api_res_sandbox = {}, {}, {}
-    resources_group_outputs, sandbox_group_outputs, extra_sandbox_group_outputs = {}, {}, {}
-
-    api_res_meta = api_res.get("meta")
-    if api_res_meta:
-        api_res_quota = api_res_meta.get("quota")
-
-    meta_group_outputs = add_outputs_from_dict(api_res_meta, meta_fields)
-    quota_group_outputs = add_outputs_from_dict(api_res_quota, quota_fields)
-
-    resources = api_res.get("resources")
-    if resources:
-        # depended on the command, the resources section can be a str list or a list that contains
-        # only one argument which is a dict
-        if type(resources[0]) == dict:
-            api_res_resources = resources[0]
-            resources_group_outputs = add_outputs_from_dict(api_res_resources, resources_fields)
-
-            sandbox = api_res_resources.get("sandbox")
-            if api_res_resources and sandbox:
-                api_res_sandbox = sandbox[0]
-                sandbox_group_outputs = add_outputs_from_dict(api_res_sandbox, sandbox_fields)
-                extra_sandbox_group_outputs = add_outputs_from_dict(api_res_sandbox, extra_sandbox_fields)
-
-                if extra_sandbox_group_outputs.get('processes'):
+    output: dict[str, Any] = dict()
+    indicator: Optional[Common.File] = None
+
+    if api_res_meta := response.get("meta", dict()):
+        output |= filter_dictionary(api_res_meta, meta_fields)
+        output |= filter_dictionary(api_res_meta.get("quota", {}), quota_fields)
+
+    if resources_list := response.get("resources"):
+        # depends on the command, the resources_list section can be a list[str] or list with a single dictionary
+        if isinstance(resources_list[0], dict):
+            resources = resources_list[0]
+            resources_group_outputs = filter_dictionary(resources, resources_fields)
+
+            if sandbox := resources.get("sandbox", [{}])[0]:  # list of single dict
+                output |= filter_dictionary(sandbox, sandbox_fields)
+                indicator = parse_indicator(sandbox, reliability)
+
+                if extra_sandbox_group_outputs := filter_dictionary(sandbox, extra_sandbox_fields):
                     for process in extra_sandbox_group_outputs.get('processes', []):
-                        if process.get('registry'):
-                            process.pop('registry')
+                        process.pop('registry', None)
+
+                    resources_group_outputs['sandbox'] = extra_sandbox_group_outputs
+                    output |= extra_sandbox_group_outputs
+        else:  # the resources section is a list of strings
+            resources_group_outputs = {"resources": response.get("resources")}
```

Comment:
Please consider moving this condition to the top + break, as that'll save a substantial indent.